File: 206-reverse-linked-list/reverse-linked-list.js
var reverseList = function(head) {
    if (head === null) {
        return null;
    }

    let nodes = [];
    let current = head;
    while (current !== null) {
        nodes.push(current);
        current = current.next;
    }

    for (let i = nodes.length - 1; i >= 0; i--) {
        nodes[i].next = i === 0 ? null : nodes[i - 1];
    }

    return nodes[nodes.length - 1];
};

File: 206-reverse-linked-list/reverse-linked-list.py
class Solution:
    def init(self, val):
        self.val = val
        self.next = None
        
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return head


        prev = None
        current = head
        while current:
            next_node = current.next
            current.next = prev
            prev = current
            current = next_node

        return prev